testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -1020,60 +937,6 @@ private void assertDeltaTrinoTableEquals(String tableName, String trinoQuery, St
                 .containsOnly(expectedRows);
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testTrinoDropColumnWithColumnMappingMode(String mode)
-    {
-        testDropColumnWithColumnMappingMode(mode, (tableName, columnName) -> onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " DROP COLUMN " + columnName));
-    }
-
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingDataProvider")
-    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testSparkDropColumnWithColumnMappingMode(String mode)
-    {
-        testDropColumnWithColumnMappingMode(mode, (tableName, columnName) -> onDelta().executeQuery("ALTER TABLE default." + tableName + " DROP COLUMN " + columnName));
-    }
-
-    private void testDropColumnWithColumnMappingMode(String mode, BiConsumer<String, String> dropColumn)
-    {
-        String tableName = "test_drop_column_" + randomNameSuffix();
-
-        onDelta().executeQuery("" +
-                "CREATE TABLE default." + tableName +
-                " (id INT, data INT, part STRING)" +
-                " USING delta " +
-                " PARTITIONED BY (part) " +
-                " LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
-                " TBLPROPERTIES ('delta.columnMapping.mode' = '" + mode + "')");
-
-        try {
-            assertThat(getTablePropertyOnDelta("default", tableName, "delta.columnMapping.maxColumnId"))
-                    .isEqualTo("3");
-            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, 10, 'part#1')");
-
-            dropColumn.accept(tableName, "data");
-            assertThatThrownBy(() -> dropColumn.accept(tableName, "part"))
-                    .hasMessageMatching("(?s).*(Cannot drop partition column: part|Dropping partition columns \\(part\\) is not allowed).*");
-            assertThat(getTablePropertyOnDelta("default", tableName, "delta.columnMapping.maxColumnId"))
-                    .isEqualTo("3");
-
-            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
-                    .containsOnly(row(1, "part#1"));
-            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName))
-                    .containsOnly(row(1, "part#1"));
-
-            // Verify adding a new column with the same name doesn't allow accessing the old data
```

Comment:
We have a slightly similar coverage for dropping and adding a new column with the same name in `io.trino.plugin.deltalake.TestDeltaLakeConnectorTest#testDropAndAddColumnShowStatsForColumnMappingMode`.

Please add though also an integration test which verifies this specific use case.